e3db/client.py

```diff
@@ -1519,16 +1560,147 @@ def read_note(self, note_id, auth_params={}, auth_headers={}) -> Note:
+
+        private_signing_key : str
+
+        auth_params : dict
+            Extra request parameters for EACP authorizations. Optional.
+
+        auth_headers : dict
+            Extra request headers for EACP authorizations. Optional
+
+        api_url: str
+            Defaults to "https://api.e3db.com"
+
+        client_id: str
+            Defaults to the empty string.
+
+        Returns
+        -------
+        e3db.Note
+            Decrypted note        
+        """
+
         auth_params['note_id'] = note_id
-        url = self.__get_url("v2", "storage", "notes")
-        response = requests.get(url=url, auth=self.e3db_tsv1_auth, params=auth_params, headers=auth_headers)
-        self.__response_check(response)
+        url = f"{api_url}/v2/storage/notes"
+
+        auth = E3DBTSV1Auth(private_signing_key, client_id)
+        response = requests.get(url=url, auth=auth, params=auth_params, headers=auth_headers)
+        Client.__response_check(response)
         note = Note.decode(response.json())
-        decrypted_note = self.decrypt_note(note, self.encryption_keys.private_key)
+        decrypted_note = Client.decrypt_note(note, private_encryption_key)
+        return decrypted_note
+
+    def read_note_by_name(self, name, auth_params={}, auth_headers={}) -> Note:
+        """
+        Public method to read a note by name. This is a wrapper for the 
+        static read_anonymous_note_by_name class to make is easier for a client to read
+        a note without looking up their keys. 
+
+        Parameters
+        ----------
+        note_id : str
+            UUID assigned by Tozstore, used to identify a note.
+
+        auth_params : dict
+            Extra request parameters for EACP authorizations. Optional.
+
+        auth_headers : dict
+            Extra request headers for EACP authorizations. Optional
+
+        Returns
+        -------
+        e3db.Note
+            Decrypted note
+        """
+        return Client.read_anonymous_note_by_name(name,
+                                                self.encryption_keys.private_key,
+                                                self.signing_keys.private_key,
+                                                auth_params,
+                                                auth_headers,
+                                                self.api_url,
+                                                self.client_id) 
+
+    @staticmethod
+    def read_anonymous_note_by_name(name: str,
+                                    private_encryption_key: str,
+                                    private_signing_key: str,
+                                    auth_params: dict={},
+                                    auth_headers: dict={},
+                                    api_url: str=DEFAULT_API_URL,
+                                    client_id: str="") -> Note:
+        """
+        Anonymously read a note by name. In the NoteOptions class the note name is
```

Comment:
see now I don't even need to get up on the soapbox of  "adding the type of a variable to a name is unclear and inefficient (especially in typed languages where the complier can tell you the type...)"